datafusion/expr/src/logical_plan/tree_node.rs

```diff
@@ -710,13 +714,12 @@ impl LogicalPlan {
             node: &LogicalPlan,
             f: &mut F,
         ) -> Result<TreeNodeRecursion> {
-            f(node)?
-                .visit_children(|| {
-                    node.apply_subqueries(|c| apply_with_subqueries_impl(c, f))
-                })?
-                .visit_sibling(|| {
-                    node.apply_children(|c| apply_with_subqueries_impl(c, f))
-                })
+            f(node)?.visit_children(|| {
+                node.apply_subqueries(|c| apply_with_subqueries_impl(c, f))?
+                    .visit_sibling(|| {
```

Comment:
i don't fully understand this -- the `.visit_sibling` call looks like it is in th `visit_children` closure now rather than being chained. Is that intended?

Reply:
Yeah, it might look a bit weird at first, but actually the subquery expression plans in a `LogicalPlan` node and the children of that node are kind of siblings (in terms of recursion) and all of them are kind of children of the node.

So the `.visit_sibling()` continuation from `.apply_subqueries()` (subquery expression plans of a node) to `.apply_children()` (children of a node) makes sense and it needs to happen in a `.visit_children()` continuation.